### DISCUSSION TITLE
Leistung setzen per DPL

### DISCUSSION THREAD

**Discussion by rojanet** [hoylabs/OpenDTU-OnBattery#1208] [General]
Hallo ich hätte mal eine Frage.

Wie wird der DPL wert Leistung gesetzt.

Non Persistent oder Persistent.

vielen Dank

**Comment by spcqike** [comment]
Non persistent. 

Persistent wäre der Flash nach wenigen Monaten defekt.

**Comment by rojanet** [comment]
Weil ich per Home Assitent per opendtuonbatterie diese werte bekomme. das kommt mir so vor als ob der das rein schreibt.

kann man irgendwie raus bekommen welchen wert er gesetzt hat per dpl.

![Screenshot 2024-08-29 19 32 54](https://github.com/user-attachments/assets/602fee90-9091-4a16-adf5-cc05689fc310)

**Reply by schlimmchen** [reply]
Die Frage, wie der DPL das limit setzt, hat @spcqike dir korrekt beantwortet.

Du fragst dich nun, warum es in deinem HA Dialog so aussieht, als sei dein Limit von 270W sowohl persistent als auch nicht-persistent gespeichert. Das sieht deshalb so aus, weil der Inverter bei seiner Antwort bzw. Info-Nachricht an die DTU keine Unterscheidung macht zwischen persistentem und nicht-persistentem Limit. Da steht einfach "das ist mein aktuelles Limit" drin.

Aber um von HA aus entscheiden zu können, ob du ein neues Limit persistent oder nicht-persistent Schreiben willst, muss es unterschiedliche Knöpfe geben. Diese Knöpfe (Eingabefelder) schreiben unterschiedliche MQTT Topics. Die werden von der DTU unterschieden und es wird das Limit entweder mit der Bitte es zu persistieren an den Inverter übermittelt, oder ohne diese Bitte.

Vorausgefüllt sind diese Eingabefelder aber mit dem aktuell vom Inverter zurückgemeldeten Limit, und zwar beide Typen gleichermaßen. Deshalb "sieht es so aus" als ob das Limit persistent gespeichert wäre. Es ist aber nur einfach gerade das Limit des Inverters.

Total schwierig das schön zu erklären, finde ich. Aber ich hoffe es wird klar.

**Reply by rojanet** [reply]
OK vielen dank. gut erklärt.

**Reply by DanielR92** [reply]
Das ist zwar korrekt, man könnte hier aber ein Workaround machen. 

Man speichert im System den letzten Pers. Limit Wert ab (im Flash). 
Damit kann in Zukunft bei Änderungen die nur im Non-Pers. betrieben werden, leichter unterscheiden. 

Bei OpenDTU-OnBattery wird es an sich ja schon sogar so gehandhabt, hier wurde ja schon die Grundlast definiert. 
Ohne im Code geschaut zu haben, gehe ich mal davon aus das dieser Wert einmalig bei einer Änderung hinterlegt worden ist (wenn dies manuell geändert wird?). *

Denn das Problem ist ja nicht nur wenn man den Energy-Meter nicht erreicht, sondern was wäre wenn man im Urlaub ist und die DTU nicht mehr reagiert? Somit würde der WR den Pers-Limit nutzen. :) 

* Ist das nicht der Fall? => Feature Request. ;)

**Reply by spcqike** [reply]
> hier wurde ja schon die Grundlast definiert.

die Grundlast ist ebenfalls ein non persistenter Wert, den der DPL einfach setzt, wenn der Stromzähler nicht erreichbar ist.

Wenn du deine opendtu aus schaltest und den Wechselrichter neustartest, startet er mit seinem persistenten Limit. Das ist nicht die Grundlast aus dem DPL.

> Ist das nicht der Fall? => Feature Request. ;)

wurde anderswo bereits diskutiert. 

Ich denke, es ist einfacher, wenn jeder Benutzer ein eigenes persistentes Limit überträgt. einmalig, ohne DPL.

Warum sollte ausgerechnet die Grundlast das persistente Limit sein? nach deinem Beispiel:

> wenn man im Urlaub ist und die DTU nicht mehr reagiert?

Was bringt es mir, wenn die Grundlast den Akku trotzdem leer saugt? Wie lange bin ich weg? Wie schnell kann ich reagieren? Im Sommer sollte das System auch mit 400W Grundlast und 5kWh Akku 24/7 für mehrere Wochen durchlaufen, da der Akku tagsüber geladen wird. Im Winter könnte es aber viel zu viel sein und der Akku nach 1-2 Tagen leer, so dass das BMS hart ausschaltet.

**Reply by DanielR92** [reply]
Ich nutze kein Akku....
